pkg/virt-handler/vm.go

```diff
@@ -192,6 +194,7 @@ func NewController(
 	migrationProxy migrationproxy.ProxyManager,
 	capabilities *nodelabellerapi.Capabilities,
 	hostCpuModel string,
+	startedServer map[string]struct{},
```

Comment:
I think this should be a struct with a context per started server, so you can stop them individually.